contracts/LinearVesting.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title LinearVesting
+ * @dev Linear vesting contract
+ */
+contract LinearVesting is Ownable {
+    IERC20 immutable token;
+    uint256 public start; // when the vesting starts
+    uint256 public cliff; // period in seconds
+    uint256 public duration; // vesting duration
+    address public receiver; // recipient
+    uint256 public amount; // number of tokens to be sent
+    uint256 public totalAmount; // total amount of tokens locked in the contract
+
+    constructor(address _token, uint256 _duration, uint256 _cliff) {
+        require(_duration > 0, "Duration must be positive");
+        require(
+            _duration >= _cliff,
+            "Cliff cannot be greater than the vesting duration"
+        );
+
+        cliff = _cliff;
+        token = IERC20(_token);
+        // start is not initialized in the constructor to ensure that the vesting period starts after the locked tokens
+        duration = _duration;
+    }
+
+    /**
+     * @dev Lock tokens for a specified period
+     * @param _receiver description receiver
+     * @param _amount number of token to lock
+     */
+    function lock(address _receiver, uint256 _amount) public onlyOwner {
+        // Check that the amount to be locked is positive
+        require(_amount > 0, "The amount to lock must be positive");
+
+        // can only be locked once
+        require(totalAmount == 0, "Already locked");
+
+        // transfer the tokens from the original owner (msg.sender) to the vesting contract (address(this)).
+        token.transferFrom(msg.sender, address(this), _amount);
+
+        receiver = _receiver;
+        amount = _amount;
+        start = block.timestamp;
+        totalAmount = 0;
+    }
```

Comment:
Based on the require, this function can be called multiple times, is it the expected behaviour ?